packages/itmat-interface/src/graphql/resolvers/jobResolvers.ts

```diff
@@ -42,22 +43,43 @@ export const jobResolvers = {
             }
 
             /* create job */
-            const job: Models.JobModels.IJobEntryForDataCuration = {
-                id: uuid(),
-                jobType: JOB_TYPE.DATA_UPLOAD,
-                studyId: args.studyId,
-                requester: requester.id,
-                requestTime: new Date().valueOf(),
-                receivedFiles: [args.file],
-                error: null,
-                status: 'QUEUED',
-                cancelled: false,
-                data: {
-                    dataVersion: args.version,
-                    versionTag: args.tag
-                }
-            };
-
+            const parts = file.fileName.split('.');
+            const dataFormat = parts[parts.length - 1];
+            let job: Models.JobModels.IJobEntryForDataCuration;
+            if (dataFormat === 'json') {
+                job = {
+                    id: uuid(),
+                    jobType: JOB_TYPE.DATA_UPLOAD_JSON,
+                    studyId: args.studyId,
+                    requester: requester.id,
+                    requestTime: new Date().valueOf(),
+                    receivedFiles: [args.file],
+                    error: null,
+                    status: 'QUEUED',
+                    cancelled: false,
+                    data: {
+                        dataVersion: args.version,
+                        versionTag: args.tag
+                    }
+                };
+            } else {
+                job = {
+                    id: uuid(),
+                    jobType: JOB_TYPE.DATA_UPLOAD_CSV,
+                    studyId: args.studyId,
+                    requester: requester.id,
+                    requestTime: new Date().valueOf(),
+                    receivedFiles: [args.file],
+                    error: null,
+                    status: 'QUEUED',
+                    cancelled: false,
+                    data: {
+                        dataVersion: args.version,
+                        versionTag: args.tag
+                    }
+                };
+            }
+            
```

Comment:
can you abstract this away since it's the same object?

something like (just an example) (you can do it with higher order function, switches, other ways if you like)
```typescript
const dataFormatToJobType = {
    json: JOB_TYPE.DATA_UPLOAD_JSON,
    csv:  JOB_TYPE.DATA_UPLOAD_CSV
};
if (!dataFormatToJobType[dataFormat]) {
    // send error
}
const job: IJob = {
                    id: uuid(),
                    jobType: dataFormatToJobType[dataFormat],
                    studyId: args.studyId,
                    requester: requester.id,
                    requestTime: new Date().valueOf(),
                    receivedFiles: [args.file],
                    error: null,
                    status: 'QUEUED',
                    cancelled: false,
                    data: {
                        dataVersion: args.version,
                        versionTag: args.tag
                    }
                };
```

Reply:
thanks, Chon. It's done.